examples/double_gyre.jl

```diff
@@ -0,0 +1,285 @@
+# # Double Gyre
+#
+# This example simulates a double gyre following:
+# https://mitgcm.readthedocs.io/en/latest/examples/baroclinic_gyre/baroclinic_gyre.html
+
+using Oceananigans
+using Oceananigans.Units
+using Oceananigans.Grids: xnode, ynode, znode
+using Oceananigans.TurbulenceClosures: VerticallyImplicitTimeDiscretization
+using CairoMakie
+using Printf
+
+const Lx = 4000kilometers # east-west extent [m]
+const Ly = 6000kilometers # north-south extent [m]
+const Lz = 1.8kilometers  # depth [m]
+
+Δt₀ = 10minutes
+stop_time = 10years
+
+Nx = 160
+Ny = 240
+Nz = 50
+
+σ = 1.2 # stretching factor
+hyperbolically_spaced_faces(k) = - Lz * (1 - tanh(σ * (k - 1) / Nz) / tanh(σ))
+
+grid = RectilinearGrid(CPU();
+                       size = (Nx, Ny, Nz),
+                          x = (-Lx/2, Lx/2),
+                          y = (-Ly/2, Ly/2),
+                          z = hyperbolically_spaced_faces,
+                   topology = (Bounded, Bounded, Bounded))
+
+# We plot vertical spacing versus depth to inspect the prescribed grid stretching:
+
+fig = Figure(resolution=(1200, 800))
+ax = Axis(fig[1, 1], ylabel = "Depth (m)", xlabel = "Vertical spacing (m)")
+lines!(ax, grid.Δzᵃᵃᶜ[1:grid.Nz], grid.zᵃᵃᶜ[1:grid.Nz])
+scatter!(ax, grid.Δzᵃᵃᶜ[1:Nz], grid.zᵃᵃᶜ[1:Nz])
```

Comment:
use `zspacings` here rather than explicitly referencing grid properties

Reply:
`zspacings`  and `znodes`